src/main/java/com/google/devtools/build/lib/skyframe/ActionMetadataHandler.java

```diff
@@ -406,7 +406,6 @@ public void injectFile(Artifact output, FileArtifactValue metadata) {
         !output.isTreeArtifact() && !output.isChildOfDeclaredDirectory(),
         "Tree artifacts and their children must be injected via injectTree: %s",
         output);
-    checkState(executionMode.get(), "Tried to inject %s outside of execution", output);
```

Comment:
As much as removing checks is scary -- I think that's the right call -- it looks like this boolean literally protects against the metadata cache only (am I missing something?). We set it right after the metadata cache check and start with false otherwise.

If the metadata handler was to hand around after, we could consider a tri-state here maybe (checking cache -> cache hit/execution), but I don't see the point.